lib/tdf3/src/client/index.ts

```diff
@@ -227,6 +231,18 @@ function asPolicy(scope: Scope): Policy {
   };
 }
 
+const fetchKasKeyWithCache = (
+  cache: Record<string, ReturnType<typeof fetchKasPublicKey>>,
+  ...params: Parameters<typeof fetchKasPublicKey>
+): ReturnType<typeof fetchKasPublicKey> => {
+  const cacheKey = JSON.stringify(params);
+  if (cacheKey in cache) {
+    return cache[cacheKey];
+  }
+  cache[cacheKey] = fetchKasPublicKey(...params);
```

Comment:
Not really sure if this is the case, but it looks like if multiple calls with the same parameters happen simultaneously, the fetchKasPublicKey is gonna be called multiple times and store multiple Promises overriding the previous one.

Might lead to a funky race conditions.